File: library/src/main/java/com/gazman/db/callbacks/MainThreadCallback.java
package com.gazman.db.callbacks;

import android.os.Handler;
import android.os.Looper;

/**
 * Created by Ilya Gazman on 7/9/2016.
 */
public abstract class MainThreadCallback<T> extends DataBaseDataCallback<T> {
    static final Handler HANDLER = new Handler(Looper.getMainLooper());

    protected MainThreadCallback() {
        super(HANDLER);
    }
}

File: src/main/java/com/gazman/db/callbacks/MainThreadCallback.kt
package com.gazman.db.callbacks

import android.os.Handler
import android.os.Looper

/**
 * Created by Ilya Gazman on 7/9/2016.
 */
abstract class MainThreadCallback<T> protected constructor() : DataBaseDataCallback<T>(HANDLER) {
    companion object {
        val HANDLER = Handler(Looper.getMainLooper())
    }
}